### DISCUSSION TITLE
Autodetect CONTRIBUTING.md files

### DISCUSSION THREAD

**Discussion by thinkverse** [community/community#22663] [New to GitHub]
<p>Hello there. 👋</p>
<p>Not sure if this is the correct place to suggest this :thinking: , but let’s do it anyway. :slightly_smiling_face:</p>
<p>Contributing files are a big part of open-source projects, it would be nice if it was automatically detected like our README’s and LICENSE files and added to the sidebar for quick and easy access.</p>
<p>Something like this maybe? :thinking:</p>
<p><img src="https://user-images.githubusercontent.com/2221746/181072534-5b1ccff5-d4ea-41fb-9906-1c428f7f8f3b.png" alt="Contributing section added to sidebar" data-base62-sha1="xdFNX0WUxSqoiFH4ExPirXj3iho" width="434" height="139"></p>
<p>What does everybody else think about this? :slightly_smiling_face:</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>We do already autodetect CONTRIBUTING files and surface them in various ways. For example, when opening a new issue or PR if you haven’t viewed the CONTRIBUTING guide (or other <a href="https://docs.github.com/en/github/building-a-strong-community/creating-a-default-community-health-file">community health files</a>) before or if they’ve been updated you’ll get a callout in the sidebar:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8cfd1f15f5c8b647debdaeed5b41b2b4c0dbc0ec.png" data-download-href="/uploads/short-url/k7f9FKEPgzu92P80DnqCStnbMde.png?dl=1" title="CleanShot 2020-09-04 at 10.35.43@2x"><img src="https://user-images.githubusercontent.com/1038121/181072535-fe30f617-dd51-4f6b-9148-3c713bc7fb82.png" alt="CleanShot 2020-09-04 at 10.35.43@2x" data-base62-sha1="k7f9FKEPgzu92P80DnqCStnbMde" width="331" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/8cfd1f15f5c8b647debdaeed5b41b2b4c0dbc0ec_2_331x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/8cfd1f15f5c8b647debdaeed5b41b2b4c0dbc0ec_2_496x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/8/8cfd1f15f5c8b647debdaeed5b41b2b4c0dbc0ec_2_662x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8cfd1f15f5c8b647debdaeed5b41b2b4c0dbc0ec_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">CleanShot 2020-09-04 at 10.35.43@2x</span><span class="informations">810×1222 70.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Additionally, even if they haven’t been updated recently, there are reminders like this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/14c5ce4b758971d5f0ef8aab97b9b0bb9ba732b0.png" data-download-href="/uploads/short-url/2XLltO9ht2kHJD9YozgEhavIRCo.png?dl=1" title="CleanShot 2020-09-04 at 10.39.01@2x"><img src="https://user-images.githubusercontent.com/1038121/181072536-c7ba4907-d8ef-4f09-9294-40fce6bb4a41.png" alt="CleanShot 2020-09-04 at 10.39.01@2x" data-base62-sha1="2XLltO9ht2kHJD9YozgEhavIRCo" width="690" height="107" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/14c5ce4b758971d5f0ef8aab97b9b0bb9ba732b0_2_690x107.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/14c5ce4b758971d5f0ef8aab97b9b0bb9ba732b0_2_1035x160.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/14c5ce4b758971d5f0ef8aab97b9b0bb9ba732b0_2_1380x214.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/14c5ce4b758971d5f0ef8aab97b9b0bb9ba732b0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">CleanShot 2020-09-04 at 10.39.01@2x</span><span class="informations">2178×338 56 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Research has shown that these kind of “just in time” reminders or notifications are often more effective than offering links in places where there’s less of a contextual relationship.</p>
<p>However, we’re always looking to improve the GitHub platform, so I’ll pass along your feedback to the appropriate team. I can’t make any promises as to if or when it might happen, but you can rest assured the right people will hear your feedback.</p>
<p>Thanks for your passion and interest in GitHub!</p>

**Reply by DeveloppeurPascal** [reply]
Those links are only visible when someone access DISCUSSION or ISSUES, it would be better to display the link on the repository like asked in #14247 to signal to everybody that contributions are welcome and how to do it.
